### DISCUSSION TITLE
How to subscribe to events and access chat history in the extension?

### DISCUSSION THREAD

**Discussion by RusDyn** [cline/cline#793] [Feature Requests]
Hello,

I am developing my own VS Code extension and would like to integrate it with Cline API. Specifically, I want to:
- Subscribe to events triggered by Cline (e.g., to know when it has finished its work and is waiting for user input).
- Access the chat history.

I’ve reviewed the API documentation but couldn’t find a way to achieve this. 

Is it possible to subscribe to such events or retrieve chat history? 
If not, are there any plans to expose this functionality in the future?

Any guidance or suggestions would be greatly appreciated!

Thank you!

**Comment by dcbartlett** [comment]
At this time, this functionality does not exist.  In order to support this, a major refactor would be needed.  The chat history is stored to disk as it it comes in, you could possibly read from disk, but they are stored in numeric generated folders.  Take a look in `Code/User/globalStorage/saoudrizwan.claude-dev/tasks`.  This will be in different places depending on your OS.  

Windows: `%appdata%\`
Linux: `~/.config/`
OSX: `/Library/Application Support/`
